### PR TITLE
React Capstone Project

### PR DESCRIPTION
**Pull request by MayPyone**
-Received data from API
-The data are shown on the home page.
-The user will be detailed page when the button is clicked.
-Add tests for components.
-Deployed the project.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,151 @@
...
+<!-- Features -->
+
+### Key Features <a name="key-features"></a>
+
+- **Different kinds of cuisines**
+- **Detailed recipes**
+- **Navigating to detail**
+
+<p align="right">(<a href="#readme-top">back to top</a>)</p>
+
```
Comment (topeogunleye) [inline README.md:47]: Hi @MayPyone,

I hope you're having a pleasant day! It appears that the video presentation, a requisite component of the project submission, is currently missing. The presentation is a valuable piece of your submission, allowing you to showcase the functionalities and features of your project and articulate aspects of your code that you find particularly intriguing or noteworthy. 🌟

### **Action Required: 🛠️**

1. **Record the Video Presentation:** Please record a video presenting the features of your project. The maximum length of the video is 5 minutes. Ensure that your presentation is clear, concise, and in professional English.
2. **Include Essential Elements:** Your video should comprise a succinct description of the project, its purpose, and a demonstration of all the features of your project. Additionally, please highlight a segment of the code that you find interesting or are proud of creating.
3. **Use Recommended Tools:** It's advised to use Loom for recording the video, but feel free to use any tool you are comfortable with as long as the quality of the video is maintained and it is easily accessible.

### **Suggestions: 🌿** [OPTIONAL]
- When recording, ensure that your voice is clear, and the video is free of unnecessary disruptions.
- Practice going over the points you want to discuss beforehand to ensure smooth delivery within the time constraint.

Once the video has been recorded, please include the link to it in your submission, and feel free to tag me or drop a comment in the PR thread so that I can review it promptly.

Looking forward to seeing your presentation and hearing about your experiences and learning while building this project! 🌟

Keep up the great work, and happy coding! 🚀

Best Regards,
Temitope

File: README.md [hunk 2]
```diff
@@ -1 +1,151 @@
...
+## 🚀 Live Demo <a name="live-demo"></a>
+
+- [Live Demo Link](https://react-capstone-project-wiu2.onrender.com/)
```
Comment (topeogunleye) [inline README.md:50]: ### **Feedback on Live Version Deployment: 🌐**

Hi @MayPyone,

I hope you’re having a productive and pleasant day! I noticed the project has been deployed and the link has been well placed in the README, as per the project requirements, which is great! 🌟 However, upon visiting the link, it seems to be returning a "Bad Gateway" error. This usually implies there might be issues with the deployment configuration or the server hosting your application. 

### **Action Required: 🛠️**
1. **Review Deployment Configuration:** Please revisit your deployment settings on Render, and ensure that all configurations are set correctly, including any environment variables that your project might be using.
   
2. **Check Console Logs:** If available, review any error logs provided by the hosting service, they usually provide insights into what might be going wrong.

3. **Re-Deploy:** Once you’ve made any necessary adjustments, redeploy the application and ensure that both the home and detail page work as expected.

4. **Update README Link:** If changes in configuration result in a new URL, please remember to update the link in the README accordingly.


I know resolving deployment issues can sometimes be challenging, but I'm confident you'll manage to get it working flawlessly! 💪 Looking forward to seeing your project live! 🚀

Once the issues are resolved, please don’t hesitate to tag me or drop a comment in the PR thread, and I will review it promptly. 

Happy Coding! 🌟

Best Regards,
Temitope

File: src/component/__test__/CuisineList.test.js [hunk 3]
```diff
@@ -0,0 +1,31 @@
+import React from 'react';
```
Comment (topeogunleye) [inline src/component/__test__/CuisineList.test.js:1]: ### **Feedback on Test Implementation: 🧪**

Hello @MayPyone,

I have reviewed your test code and it is great to see your attention to detail in ensuring that the components are thoroughly tested! 🌟 The test case you've written for the `CuisineList` component is coherent and succinct. It’s good practice to verify different elements in the component like you did with text, image attributes, and the presence of other elements, which aligns well with the project requirements.

### **Action Required: 🛠️**
1. **More Test Cases**: The project requirements specify unit tests for several components and some pure functions in the Redux code. I encourage you to continue writing more test cases for other components and Redux functions to ensure extensive coverage.
2. **Integration Tests**: While the provided test is more of a unit test, considering adding some integration tests where you test the interaction between different components, and potentially mock data fetching to simulate different states of your app.
3. **Redux Store Connection Test**: Consider adding tests to ensure the correct connection and interaction with the Redux store, as it is also part of the project requirements.'
4. **Failed Test**: It seems like there is an error causing the tests to fail. The encountered error, `Jest encountered an unexpected token` usually indicates that your code or its dependencies use non-standard JavaScript syntax, or when Jest is not configured to support such syntax

### **Summary: 📜**
In conclusion, you have made commendable progress! 🚀 You have a strong foundation with your current tests, and building on it by adding more extensive test cases will ensure the robustness of your application. Keep up the good work, and looking forward to seeing more comprehensive tests covering all aspects of your app! 💪

Happy Testing! 🧪

Best Regards,
Temitope

File: src/component/Detail.js [hunk 4]
```diff
@@ -0,0 +1,65 @@
+import './home.css';
+import React, { useEffect } from 'react';
+import { v4 as uuidv4 } from 'uuid';
+import { useSelector, useDispatch } from 'react-redux';
+import { Card } from 'react-bootstrap';
+import { IoIosArrowBack } from 'react-icons/io';
+import { NavLink } from 'react-router-dom';
+import { fetchDetail } from '../redux/cuisine/detailSlice';
+
+export default function Detail() {
```
Comment (topeogunleye) [inline src/component/Detail.js:10]: Nice job so far😉. Let's try to ensure our project matches the given design. take note of the highlighted components. also, do well to stay consistent with the color as done in design 😉. You should use the same lighter shade of blue used on your home page i.e `#6b8ecf` to keep the colors consistent as seen in the behance design.  , instead of making use of white. do well to implement that.

image

**Review by topeogunleye — CHANGES_REQUESTED** [review]
Hi @MayPyone, 

Good job so far!
There are some issues that you still need to work on to prepare your project for the final evaluation but you are almost there!

## Suggested changes
## (Highlights) Good Points: 

Creative Commons Attribution 👏
You've done an excellent job in giving proper credit to Nelson Sakwa, the original designer. The "Acknowledgements" section in the README adheres to the Creative Commons License requirements. Well done!


Page Structure and Navigation 🗂️
Your project structure is organized logically, making it easy to navigate through your code. Your use of react-router-dom to handle page routing without reloads fits perfectly within the project requirements for a Single Page Application (SPA).

Technical Requirements 🛠️
Your implementation of Redux for state management is spot-on. You've correctly stored API data in the Redux store and connected it to your components. The filter component works seamlessly, adding a nice touch to the user experience.

_Check the comments under the review._

You can use as many of my suggestions as you want. If there is anything you would like to skip - feel free to do that. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- Using "feature" as a branch name is not descriptive enough and can make it difficult for other developers to understand what changes were made in the branch. For future pull requests, please consider using a more descriptive branch name that reflects the changes made in the branch.
Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

**Review by V-Blaze — APPROVED** [review]
Hi @MayPyone  , 

# Approved 🟢 🟢 

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 
giphy

## Highlights

- The presentation video is clear and meets all requirements 💯
-  Deployed project works as expected👍
-  All test cases are passing🟢 👍

## Optional suggestions

_Every comment with the [OPTIONAL] prefix won't stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better. Some of them were simply missed by the previous reviewer and addressing them will really improve your application._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._
